iothub/device/src/Common/Exceptions/IotHubCommunicationException.cs

```diff
@@ -7,8 +7,14 @@
 namespace Microsoft.Azure.Devices.Client.Exceptions
 {
     /// <summary>
-    /// The exception that is thrown when an attempt to communicate with the IoT Hub service fails.
+    /// This exception is thrown when an attempt to communicate with the IoT Hub service fails due to transient
```

Comment:
After retries have exhausted?

Reply:
We have indefinite retries by default in the DeviceClient unless the customers changes the retry policy. In that case, they will have to recreate the client. Do you want me to add those details?

Reply:
I think it was unclear if I'd get the exception and a retry would be happening automatically for me, or if I'd only get the exception after retries are exhausted.

I assume the latter. In that case, yes I also think it would be worth mentioning that, unless overridden, the SDK will retry indefinitely.

Reply:
@drwill-ms - Updated the messaging.